Adding contribution guide and enable copyright checks as a pre-commit hook (#91)
graphcore-research/pyscf-ipu: CONTRIBUTING.md

We hope this is useful for the generation of large-scale datasets needed
for training machine-learning models. We are interested in hearing any and
all feedback so feel free to raise any questions, bugs encountered, or enhancement requests as [Issues](https://github.com/graphcore-research/pyscf-ipu/issues).

## Setting up a development environment
We recommend using the conda package manager as this can automatically enable
the Graphcore Poplar SDK.  This is particularly useful in VS Code which can automatically
activate the conda environment in a variety of scenarios:
* visual debugging
* running quick experiments in an interactive Jupyter window
* using VS code for Jupyter notebook development.

The following assumes that you have already set up an install of conda and that
the conda command is available on your system path.  Refer to your preferred conda
installer:
* [miniforge installation](https://github.com/conda-forge/miniforge#install)
* [conda installation documentation](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html).

> 1. Create a new conda environment with the same python version as your OS.

Author: ```suggestion
1. Create a new conda environment with the same python version as required by the Poplar SDK.
```

# Contributing to pyscf-ipu

This project is still evolving but at the moment is focused around a high-performance and easily hackable implementation of Gaussian basis set DFT.
We hope this is useful for the generation of large-scale datasets needed
for training machine-learning models. We are interested in hearing any and
all feedback so feel free to raise any questions, bugs encountered, or enhancement requests as [Issues](https://github.com/graphcore-research/pyscf-ipu/issues).

## Setting up a development environment
We recommend using the conda package manager as this can automatically enable
the Graphcore Poplar SDK.  This is particularly useful in VS Code which can automatically
activate the conda environment in a variety of scenarios:
* visual debugging
* running quick experiments in an interactive Jupyter window
* using VS code for Jupyter notebook development.

> The following assumes that you have already setup an install of conda and that

Reviewer: ```suggestion
The following assumes that you have already set up an install of conda and that
```

the conda command is available on your system path.  Refer to your preferred conda
installer:
* [miniforge installation](https://github.com/conda-forge/miniforge#install)
* [conda installation documentation](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html).

> 1. Create a new conda environment with the same python version as you OS.

Reviewer: Really? Doesn't it have to match Poplar?

> 1. Create a new conda environment with the same python version as you OS.

Author: Yes, they are made to be the same but it would be better to say that it should match the Poplar SDK python version.

   For example, on ubuntu 20 use `python=3.8.10`
   ```bash
   conda create -n pyscf-ipu python=3.8.10
   ```

> 2. Confirm that you have the Poplar SDK installed on your machine and store the location
>    in an environment variable.  The following will test that the SDK is found and
>    configured correctly:
>    ```bash
>    export POPLAR_SDK=/path/to/sdk
>    source $POPLAR_SDK/enable
>    gc-monitor
>    ```

Reviewer: ```suggestion
2. Confirm that you have the Poplar SDK installed on your machine and store the location
   in a temporary shell variable.  The following will test that the SDK is found and
   configured correctly:
   ```bash
   TMP_POPLAR_SDK=/path/to/sdk
   source $TMP_POPLAR_SDK/enable
   gc-monitor
   ```

```

> 3. Activate the environment and store a persistent environment variable for the
>    location of the downloaded Poplar SDK. This assumes that
>    you have already downloaded the Poplar SDK.  The following example uses an
>    environment variable `$POPLAR_SDK` to store the root folder for the SDK.
>    ```bash
>    conda activate pyscf-ipu
>    conda env config vars set POPLAR_SDK=$POPLAR_SDK

Reviewer: ```suggestion
3. Activate the environment and make POPLAR_SDK a persistent environment variable.
   ```bash
   conda activate pyscf-ipu
   conda env config vars set POPLAR_SDK=$TMP_POPLAR_SDK
```

   ```

> 4. You have to reactivate the conda environment to use the `$POPLAR_SDK`
>    variable the environment.

Reviewer: ```suggestion
4. You have to reactivate the conda environment to use the `$POPLAR_SDK`
   variable in the environment.
```

   ```bash
   conda deactivate
   conda activate pyscf-ipu
   ```

5. Setup the conda environment to automatically enable the Poplar SDK whenever
   the environment is activated.
   ```bash
   mkdir -p $CONDA_PREFIX/etc/conda/activate.d
   echo "source $POPLAR_SDK/enable" > $CONDA_PREFIX/etc/conda/activate.d/enable.sh
   ```

> 6. Check that everything is working by once again reactivating the pyscf-ipu
>    environment and calling `gc-monitor`:
>    ```bash
>    conda deactivate
>    conda activate pyscf-ipu
>    gc-monitor
>    ```

Reviewer: ```suggestion
6. Check that everything is working by once reactivating the pyscf-ipu
   environment _in a new shell_ and calling `gc-monitor`:
   ```bash
   conda deactivate
   conda activate pyscf-ipu
   gc-monitor
   ```
```
